pip_audit/dependency_source/requirement.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+from typing import Iterator, List
+
+from packaging.requirements import Requirement
+from pip_api import parse_requirements
+from pip_api.exceptions import PipError
+
+from pip_audit.dependency_source import (
+    DependencyResolver,
+    DependencyResolverError,
+    DependencySource,
+    DependencySourceError,
+)
+from pip_audit.service import Dependency
+
+
+class RequirementSource(DependencySource):
+    def __init__(self, filenames: List[Path], resolver: DependencyResolver):
+        self.filenames = filenames
+        self.resolver = resolver
+
+    def collect(self) -> Iterator[Dependency]:
+        # TODO(alex): I wonder whether we need to do some deduplication of requirements/dependencies
```

Comment:
I have an idea that if two requirements files share the same dependency, it might appear twice in the output and get audited twice.

If the versions are different, I think you'd want two separate entries since they can suffer from different vulnerabilities. But in the case of identical versions, it might be worth deduplicating them.

Reply:
Yeah, deduplicating them makes sense.

I wonder if it also make sense to warn or otherwise signal users when they have two different (and incompatible) versions for the same dependency -- actually installing those into the same (virtual) environment would be a conflict, one that `pip-audit` manages to avoid circumstantially (by using a different venv for each requirement source). 

But that's perhaps too far out of scope, so we can ignore it for now 🙂